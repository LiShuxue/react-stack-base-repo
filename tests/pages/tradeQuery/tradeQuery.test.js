import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TradeQuery from 'src/pages/tradeQuery';

test('should render TradeQuery page correctly', () => {
  const testRenderer = renderer.create(<TradeQuery />);
  const dom = testRenderer.toJSON();
  expect(dom).toMatchSnapshot();
});

test('should have "default button" and "disabled button"', () => {
  const { getByText } = render(<TradeQuery />);

  expect.assertions(2);

  const defaultBtn = getByText('default Button');
  expect(defaultBtn).toBeInTheDocument();

  const disabledBtn = getByText('disabled Button');
  expect(disabledBtn).toBeInTheDocument();
});

test('should have a datalist', () => {
  const { getByTestId } = render(<TradeQuery />);
  const datalist = getByTestId('datalist');
  expect(datalist).toBeInTheDocument();
});
