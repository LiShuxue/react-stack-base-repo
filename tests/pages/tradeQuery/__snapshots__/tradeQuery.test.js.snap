// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TradeQuery page correctly 1`] = `
<div>
  <h1>
    TradeQuery
  </h1>
  <hr />
  <button
    className="ant-btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      default Button
    </span>
  </button>
  <br />
  <br />
  <button
    className="ant-btn"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <span>
      disabled Button
    </span>
  </button>
  <hr />
  <div
    className="ag-theme-balham-dark"
    data-testid="datalist"
    style={
      Object {
        "height": "150px",
        "width": "650px",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  </div>
</div>
`;
